wamr-compiler/CMakeLists.txt

```diff
@@ -45,6 +45,7 @@ add_definitions(-DWASM_ENABLE_DUMP_CALL_STACK=1)
 add_definitions(-DWASM_ENABLE_PERF_PROFILING=1)
 add_definitions(-DWASM_ENABLE_LOAD_CUSTOM_SECTION=1)
 add_definitions(-DWASM_ENABLE_LIB_WASI_THREADS=1)
+add_definitions(-DWASM_ENABLE_GC=1)
```

Comment:
According to the design, we want to add control to enable gc or enable gc binaryen, the change should be like:
```cmake
if (WAMR_BUILD_GC_BINARYEN EQUAL 1)
  add_definitions (-DWASM_ENABLE_GC_BINARYEN=1)
else ()
  add_definitions (-DWASM_ENABLE_GC=1)
endif ()
```
Then we can build wamrc for gc/gc_binaryen version with: `cmake -DWAMR_BUILD_GC_BINARYEN=0/1`